bcipy/task/paradigm/rsvp/copy_phrase.py

```diff
@@ -612,33 +611,75 @@ def write_session_data(self) -> None:
             _save_session_related_data(self.session_save_location,
                                        self.session.as_dict())
 
-    def write_offset_trigger(self, trigger_file: TextIO):
+    def write_offset_trigger(self) -> None:
         """Append the offset to the end of the triggers file.
-
-        Parameters
-        -----------
-        - trigger_file : open file in which to write
         """
         if self.daq.is_calibrated:
-            _write_triggers_from_inquiry_copy_phrase(
-                ['offset', self.daq.offset(self.rsvp.first_stim_time)],
-                trigger_file,
-                self.copy_phrase,
-                self.spelled_text,
-                offset=True)
-
-    def write_trigger_data(self, stim_times: List[Tuple[str, float]],
-                           trigger_file: TextIO) -> None:
+            self.trigger_handler.add_triggers(
+                [Trigger(
+                    'daq_sample_offset',
+                    TriggerType.SYSTEM,
+                    # to help support future refactoring or use of lsl timestamps only
+                    # we write only the sample offset here
+                    self.daq.offset(self.rsvp.first_stim_time)
+                )])
+        self.trigger_handler.close()
+
+    def write_trigger_data(self, stim_times: List[Tuple[str, float]], target_stimuli: str) -> None:
         """Save trigger data to disk.
 
         Parameters
         ----------
         - stim_times : list of (stim, clock_time) tuples
-        - trigger_file : data will be appended to this file
+        - target_stimuli : current target stimuli
         """
-        _write_triggers_from_inquiry_copy_phrase(stim_times, trigger_file,
-                                                 self.copy_phrase,
-                                                 self.spelled_text)
+        if self.first_run and self.daq.is_calibrated:
+            # write offset first
+            self.trigger_handler.add_triggers(
+                [Trigger(
+                    'starting_offset',
+                    TriggerType.OFFSET,
+                    # offset will factor in true offset and time relative from beginning
+                    (self.daq.offset(self.rsvp.first_stim_time) - self.rsvp.first_stim_time)
+                )]
+            )
+            self.first_run = False
+
+        triggers = self.convert_timing_triggers(stim_times, target_stimuli)
+        self.trigger_handler.add_triggers(triggers)
+
+    def convert_timing_triggers(self, timing: List[tuple], target_stimuli: str) -> List[Trigger]:
```

Comment:
This could also be made a function.

Reply:
What do you think about splitting this into two functions:

```
def trigger_type(symbol: str, target_stimuli: str) -> TriggerType:
    if symbol == 'inquiry_preview':
        return TriggerType.PREVIEW
    if 'bcipy_key_press' in symbol:
        return TriggerType.EVENT
    if symbol == '+':
        return TriggerType.FIXATION
    if target_stimuli == symbol:
        return TriggerType.TARGET
    return TriggerType.NONTARGET

def convert_timing_triggers(timing: List[tuple], target_stimuli: str) -> List[Trigger]:
    """Convert Stimuli Times to Triggers.

    Using the stimuli presentation times provided by the display, convert them into BciPy Triggers.
    """
    return [
        Trigger(symbol, trigger_type(symbol, target_stimuli), time)
        for symbol, time in timing
    ]
```

Reply:
I think this is a nice suggestion